examples/router-conversion.ts

```diff
@@ -0,0 +1,223 @@
+            properties: [
+              j.objectProperty.from({
+                key: j.identifier(identifier),
+                value: j.identifier(identifier),
+                shorthand: true,
+              }),
+            ],
+          }),
+          init: j.callExpression(j.identifier("useLocationContext"), []),
+        }),
+      ]),
+    );
+}
+
+export const parser = "tsx";
+// converts imports from react-router-dom to state/location
+export default function transformer(file: FileInfo, api: API) {
+  if (
+    file.path.includes(".test") ||
+    file.path.includes(".spec") ||
+    file.path.includes("__tests__") ||
+    file.path.includes("state/locatin")
+  ) {
+    return file.source;
+  }
+
+  try {
+    const j = api.jscodeshift;
+
+    const root = j(file.source);
+
+    const importedIdentifiers = collectImportElementsFromReactRouter(root, j);
+
+    // This file doesnt have any imports to react-router-dom, so we can skip
+    if (importedIdentifiers.length === 0) {
+      return;
+    }
+
+    function addImport(importNode: any, identifier: string) {
+      const hasImport = importNode.value.specifiers.some(
+        (p) => p.imported.name === identifier,
+      );
+
+      if (!hasImport) {
+        importNode.value.specifiers.push(
+          j.importSpecifier(j.identifier(identifier)),
+        );
+      }
+    }
+
+    for (const [importPath, identifier] of importedIdentifiers) {
+      switch (importPath) {
+        case "BrowserRouter":
+        case "Routes":
+        case "Route":
+        case "useSearchParams":
+          throw Error(`cant convert ${importPath}`);
+
+        case "Link":
+          const importNode = findOrCreateImport(root, j, "components/link");
+          const hasImport = importNode.value.specifiers.some(
+            (p) => p.local?.name === identifier,
+          );
+
+          if (!hasImport) {
+            importNode.value.specifiers.push(
+              j.importDefaultSpecifier(j.identifier(identifier)),
+            );
+          }
+          break;
+        case "Navigate":
+          addImport(findOrCreateImport(root, j, "state/location"), "Navigate");
+          break;
+        case "useMatch":
+          addImport(findOrCreateImport(root, j, "state/location"), "useMatch");
+          break;
+        case "useParams":
+          addImport(findOrCreateImport(root, j, "state/location"), "useParams");
+          break;
+        case "useNavigate":
+          addImport(
+            findOrCreateImport(root, j, "state/location"),
+            "useLocationContext",
+          );
+          handleUseLocationContextNode(root, j, "useNavigate", "navigate");
+          break;
+        case "useLocation":
+          addImport(
+            findOrCreateImport(root, j, "state/location"),
+            "useLocationContext",
+          );
+          handleUseLocationContextNode(root, j, "useLocation", "location");
+          break;
+      }
+    }
+
+    return root.toSource();
```

Comment:
`toSource` accepts an object, e.g.

```suggestion
    return root.toSource({ quote: 'single', trailingComma: true });
```